File: todo-oak-api/src/utils/response.ts

export interface ApiResponse{
  status: number,
  statusText: string,
  payload: any
}

export function respOK(data:any, status=200, statusText="OK"):ApiResponse{
  return {
    status,
    statusText,
    payload: data
  }
}

export function respErr(status=500, statusText="Internal error", data:any=null):ApiResponse{
  return {
    status,
    statusText,
    payload: data
  }
}

File: todo-fast-api/api/route/response.py

def respOK(data,status=200,statusText="OK"):
  resp={
    "status":status,
    "statusText": statusText,
    "payload": data
  }
  return resp


def respErr(statusText="Server error",status=500,data=None):
  resp={
    "status":status,
    "statusText": statusText,
    "payload": data
  }
  return resp
